### PR TITLE
Adding serialization capability. 

### PR DESCRIPTION
**Pull request by haochenuw**
- Added support to serialize RqPolys and FVCiphertexts into Vec<u8> using the to_bytes function. 
- Added deserialization support through the API  FV::from_bytes(& Vec<u8>).

### CODE REVIEW

File: src/serialize.rs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    let mut v = vec![0; fv.n];
+    for i in 0..fv.n {
+        v[i] = i as u8;
+    }
+    // encrypt v
+    let ct = fv.encrypt(&v, &pk);
+    let bytes = ct.to_bytes();
+    let ct_deserialized = FVCiphertext::from_bytes(&bytes);
+    assert_eq!(ct_deserialized, ct);
+  }
```
Comment (shubho) [inline src/serialize.rs:112]: Remove the commented part below if not needed?

**Review by shubho — APPROVED** [review]
LGTM - small change requested to delete commented lines if not used

File: src/serialize.rs (hunk 1 above)
Comment (haochenuw) [reply src/serialize.rs:112]: Removed. Thanks!

File: src/rqpoly.rs [hunk 2]
```diff
@@ -9,7 +9,7 @@ use std::sync::Arc;
 /// Holds the context information for RqPolys, including degree n, modulus q, and optionally precomputed
 /// roots of unity for NTT purposes.
 #[derive(Debug)]
-pub(crate) struct RqPolyContext<T> {
+pub struct RqPolyContext<T> {
```
Comment (haochenuw) [inline src/rqpoly.rs]: Probably doesn't need to be pub?

**Pull request merged**